Formula/c/cuda@12.9.1.rb

```diff
@@ -0,0 +1,66 @@
+class CudaAT1291 < Formula
+  # CUDA version variables
+  CUDA_VERSION = "12.9.1".freeze
+  CUDA_BUILD = "575.57.08".freeze
+  GCC_VERSION = "14".freeze
+
+  desc "NVIDIA's GPU programming toolkit"
+  homepage "https://developer.nvidia.com/cuda-toolkit"
+
+  on_arm do
+    CUDA_SUFFIX = "_sbsa".freeze
+    CUDA_RUN_SHA256 = "64f47ab791a76b6889702425e0755385f5fa216c5a9f061875c7deed5f08cdb6".freeze
+  end
+  on_intel do
+    CUDA_SUFFIX = "".freeze
+    CUDA_RUN_SHA256 = "0f6d806ddd87230d2adbe8a6006a9d20144fdbda9de2d6acc677daa5d036417a".freeze
+  end
+
+  url "https://developer.download.nvidia.com/compute/cuda/#{CUDA_VERSION}/local_installers/cuda_#{CUDA_VERSION}_#{CUDA_BUILD}_linux#{CUDA_SUFFIX}.run", using: :nounzip
+  sha256 CUDA_RUN_SHA256
+  version CUDA_VERSION
+
+  # https://docs.nvidia.com/cuda/eula/index.html
+  license :cannot_represent
```

Comment:
Is it under the Debian open source definition? Otherwise we won't ship this

Reply:
But is it open source by the Debian definition? Because otherwise it should be a cask.

Reply:
Sorry, I don't know Debian's definition of open source. It's not open source according to the common definition of open source. These are the guidelines for distributing the toolkit. https://docs.nvidia.com/cuda/eula/index.html#cuda-toolkit-supplement-to-software-license-agreement-for-nvidia-software-development-kits

I guess it should be a cask (I didn't know about those)... but I don't see anyway to customize the installation. Can you provide an example cask that does something similar?

Reply:
This doesn't meet the [OSI definition](https://opensource.org/osd) of open source. This is not acceptable for Homebrew/core, but you can try to submit upstream's prebuilt binaries to Homebrew/cask. Of note: things in Homebrew/core cannot depend on things in Homebrew/cask, so keep that in mind if you're trying to use it for another formula.